item-impl/src/main/java/com/example/auction/item/impl/PItemEntity.java

```diff
@@ -50,6 +53,14 @@ private Behavior created(PItemState state) {
 
         builder.setReadOnlyCommandHandler(GetItem.class, this::getItem);
 
+        builder.setCommandHandler(UpdateItem.class, (updateItem, ctx) -> {
+                    ItemUpdated event = ItemUpdated.from(updateItem);
```

Comment:
I wonder if there's any additional validation we should do on the item passed in.

For example:

- We should reject the command if the creator has changed
- We should not allow the status to change via this command (or, if we do, we need to be sure to trigger the appropriate behavior)
- Clients should not be able to change start/end dates or winner via this method, either
- If _none_ of the fields have changed, it should not emit an event

Some of these are handled currently by simply not copying the fields across, but this has the drawback that the client does not know which fields that it sent in the request actually changed, or that it sent a (partially) invalid request. This could cause a client to get out of date with the server.

This is another good reason to include all of the details from the original item on the request (so we have all of the fields available to validate) and to return the current state of the item in the response (so that clients can resynchronize their local state if necessary).

(Arguably, a lot of this should apply to create as well.)